lib/prometheus/move_collector.rb

```diff
@@ -10,8 +10,46 @@ def type
   end
 
   def metrics
-    move_counter = PrometheusExporter::Metric::Counter.new('move_count', 'Number of moves in the system')
-    move_counter.observe(Move.count)
-    [move_counter]
+    metric = PrometheusExporter::Metric::Gauge.new('move_gauge', 'Number of moves in the system')
+
+    statuses = [nil] + Move.statuses.values
+    suppliers = [nil] + Supplier.all
+
+    statuses.each do |status|
+      suppliers.each do |supplier|
+        observe_moves_count_buckets(metric, status: status, supplier: supplier)
+      end
+    end
+
+    [metric]
+  end
+
+private
+
+  def observe_moves_count_buckets(metric, status: nil, supplier: nil)
+    observe_moves_count(metric, status: status, supplier: supplier, date_from_offset: nil, date_to_offset: nil) # all time
+    observe_moves_count(metric, status: status, supplier: supplier, date_from_offset: nil, date_to_offset: 0) # the past
+    observe_moves_count(metric, status: status, supplier: supplier, date_from_offset: -29, date_to_offset: 0) # the past 30 days
+    observe_moves_count(metric, status: status, supplier: supplier, date_from_offset: -6, date_to_offset: 0) # the past week
+    observe_moves_count(metric, status: status, supplier: supplier, date_from_offset: -1, date_to_offset: -1) # yesterday
+    observe_moves_count(metric, status: status, supplier: supplier, date_from_offset: 0, date_to_offset: 0) # today
+    observe_moves_count(metric, status: status, supplier: supplier, date_from_offset: 1, date_to_offset: 1) # tomorrow
+    observe_moves_count(metric, status: status, supplier: supplier, date_from_offset: 0, date_to_offset: 6) # the next 7 days
+    observe_moves_count(metric, status: status, supplier: supplier, date_from_offset: 0, date_to_offset: 29) # the next 30 days
+    observe_moves_count(metric, status: status, supplier: supplier, date_from_offset: 0, date_to_offset: nil) # the future
+  end
+
+  def observe_moves_count(metric, status: nil, supplier: nil, date_from_offset: nil, date_to_offset: nil)
+    # TODO: investigate how to move these queries to a read-only replica database
```

Comment:
This is interesting: https://guides.rubyonrails.org/active_record_multiple_databases.html#using-manual-connection-switching

Reply:
thanks, very interesting reading 👍

We do have read-replicas set up in most environments (not UAT), but more work will be required to make them accessible in the environments. And also need to see how well Rails behaves with multiple databases. I'll do that as a separate PR in due course.